assets/aws/update-ami-ids.sh

```diff
@@ -84,27 +85,38 @@ if [[ "${MODE}" == "cloudformation" ]]; then
     fi
     # replace AMI ID in place
     for REGION in ${REGIONS//,/ }; do
-        OLD_AMI_ID=$(grep $REGION $CLOUDFORMATION_PATH | sed -n -E "s/$REGION: \{HVM64 : (ami.*)\}/\1/p" | tr -d " ")
+        OLD_AMI_ID=$(grep "${REGION}" "${CLOUDFORMATION_PATH}" | sed -n -E "s/$REGION: \{HVM64 : (ami.*)\}/\1/p" | tr -d " ")
         NEW_AMI_ID=${IMAGE_IDS[$REGION]}
-        sed -i -E "s/$REGION: \{HVM64 : ami(.*)\}$/$REGION: \{HVM64 : $NEW_AMI_ID\}/g" $CLOUDFORMATION_PATH
+        sed -i -E "s/$REGION: \{HVM64 : ami(.*)\}$/$REGION: \{HVM64 : $NEW_AMI_ID\}/g" ${CLOUDFORMATION_PATH}
         echo "[${TYPE}: ${REGION}] ${OLD_AMI_ID} -> ${NEW_AMI_ID}"
     done
     # update version number
-    sed -i -E "s/# All AMIs from AWS - gravitational-teleport-ami-(.*)/# All AMIs from AWS - gravitational-teleport-ami-${TYPE}-${VERSION}/g" $CLOUDFORMATION_PATH
+    sed -i -E "s/# All AMIs from AWS - gravitational-teleport-ami-(.*)/# All AMIs from AWS - gravitational-teleport-ami-${TYPE}-${VERSION}/g" ${CLOUDFORMATION_PATH}
 elif [[ "${MODE}" == "terraform" ]]; then
-    TERRAFORM_PATH=../../examples/aws/terraform/README.md
+    TERRAFORM_SUBDIR="../../examples/aws/terraform"
+    TERRAFORM_PATH="${TERRAFORM_SUBDIR}/AMIS.md"
+    # get a list of non-hidden directories one level under the terraform directory (one for each of our different terraform modes)
+    pushd ${TERRAFORM_SUBDIR}
+    TERRAFORM_MODES="$(find . -mindepth 1 -maxdepth 1 -type d -not -path '*/\.*' -printf '%P\n' | xargs)"
+    popd
     if [[ "${TYPE}" == "oss" ]]; then
         TYPE_STRING="OSS"
     elif [[ "${TYPE}" == "ent" ]]; then
         TYPE_STRING="Enterprise"
     elif [[ "${TYPE}" == "ent-fips" ]]; then
         TYPE_STRING="Enterprise FIPS"
     fi
+    # change version numbers in TF_VAR_ami_name strings
+    # shellcheck disable=SC2086
```

Comment:
nit: I think shellcheck is nudging us to use bash arrays instead of strings with multiple words

Reply:
Yeah, I agree - it just seems a lot of extra effort to refactor something that works absolutely fine as-is. I'll try and use bash arrays going forward.